core/src/main/java/feign/Contract.java

```diff
@@ -282,11 +282,26 @@ protected boolean processAnnotationsOnParameter(MethodMetadata data, Annotation[
           checkState(data.headerMapIndex() == null, "HeaderMap annotation was present on multiple parameters.");
           data.headerMapIndex(paramIndex);
           isHttpAnnotation = true;
+        } else {
+          CustomParam customParam = findCustomParam(annotation);
```

Comment:
NOTE: This checks for extensions of `@CustomParam` (as explained in the initial pull request comment) as well as the simple `annotation instanceof CustomParam`, but it does not search recursively, as I don't believe it makes sense to extend an extension. This is the only thing that my untrained eye can see in this pull request that might be "iffy".

Of course, please let me know if this (or something I didn't see) stands out as needing changes. I would love to see some sort of custom param encoder (such as this) put into feign (and from some of the other open issues/pull requests, I can see that I'm not alone).